tests/drivers/random/randomnumgen_basic_api/src/main.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <random.h>
+#include <ztest.h>
+
+/*
+ * @addtogroup t_randomnumgen_basic_api
+ * @{
+ * @defgroup t_random_get_entropy test_random_get_entropy
+ * @brief TestPurpose: verify Random number generator works
+ * @details
+ * - Test Steps
+ *   -# Read random numbers from RNG driver.
+ *   -# Verify whether buffer overflow occured or not.
+ *   -# Verify whether buffer completly filled or not.
+ * - Expected Results
+ *   -# Random number should be generated.
+ * @}
+ */
+
+#define BUFFER_LENGTH 10
+
+static int recheck_random_entropy(struct device *dev, char *buffer, char num)
+{
+	int ret, i;
+	int count = 0;
+
+	for (i = 0; i <= BUFFER_LENGTH-1; i++) {
+		buffer[i] = num;
+	}
+
+	ret = random_get_entropy(dev, buffer, BUFFER_LENGTH-1);
+	if (ret) {
+		TC_PRINT("Error: random_get_entropy failed: %d\n", ret);
+		return TC_FAIL;
+	}
+	if (buffer[BUFFER_LENGTH-1] != num) {
+		TC_PRINT("Error: random_get_entropy buffer overflow\n");
+		return TC_FAIL;
+	}
+
+	for (i = 0; i < BUFFER_LENGTH-1; i++) {
+		TC_PRINT("  0x%02x\n", buffer[i]);
+		if (buffer[i] == num)
+			count++;
+	}
+	if (count >= 2)
+		return TC_FAIL;
+	else
+		return TC_PASS;
+}
+
+/*
+ * Function invokes the get_entropy callback in driver
+ * to get the random data and fill to passed buffer
+ */
+static int get_entropy(void)
+{
+	struct device *dev;
+	u8_t buffer[BUFFER_LENGTH] = {0};
+	int ret, count = 0;
+
+	TC_PRINT("Random Example! %s\n", CONFIG_ARCH);
+	dev = device_get_binding(CONFIG_RANDOM_NAME);
+	if (!dev) {
+		TC_PRINT("error: no random device\n");
+		return TC_FAIL;
+	}
+
+	TC_PRINT("random device is %p, name is %s\n",
+			dev, dev->config->name);
+
+	/* The BUFFER_LENGTH-1 is used so the driver will not
+	 * write the last byte of the buffer. If that last
+	 * byte is not 0 on return it means the driver wrote
+	 * outside the passed buffer, and that should never
+	 * happen.
+	 */
+	ret = random_get_entropy(dev, buffer, BUFFER_LENGTH-1);
+	if (ret) {
+		TC_PRINT("Error: random_get_entropy failed: %d\n", ret);
+		return TC_FAIL;
+	}
+	if (buffer[BUFFER_LENGTH-1] != 0) {
+		TC_PRINT("Error: random_get_entropy buffer overflow\n");
+		return TC_FAIL;
+	}
+
+	for (int i = 0; i < BUFFER_LENGTH-1; i++) {
+		TC_PRINT("  0x%02x\n", buffer[i]);
+		if (buffer[i] == 0)
+			count++;
+	}
+
+	/* Check whether 20% or more of buffer still filled with default
+	 * value, if yes then recheck again by filling nonzero value(0xa5)
+	 * to buffer. Recheck random_entropy and verify whether 20% or more
+	 * of buffer filled with value(0xa5) or not.
+	 */
+	if (count >= 2) {
+		ret = recheck_random_entropy(dev, buffer, 0xa5);
+		return ret;
+	}
+	TC_PRINT("PROJECT EXECUTION SUCCESSFUL\n");
```

Comment:
Do we need to print this by ourselves? I thought this would be printed automatically by ztest.

Reply:
this should not be printed manually.
look to other test cases to see what to do.
I think ztest prints this once everything is done?